pkg/e2e/operators/routemonitor.go

```diff
@@ -0,0 +1,126 @@
+}
+
+var _ = ginkgo.Describe(routeMonitorOperatorTestName, func() {
+	const (
+		operatorNamespace = "openshift-route-monitor-operator"
+		operatorName      = "route-monitor-operator"
+		operatorLockFile  = "route-monitor-operator-lock"
+
+		defaultDesiredReplicas int32 = 1
+
+		consoleNamespace = "openshift-console"
+		consoleName      = "console"
+	)
+
+	var clusterRoles = []string{
+		"route-monitor-operator-admin",
+		"route-monitor-operator-edit",
+		"route-monitor-operator-view",
+	}
+
+	h := helper.New()
+
+	checkClusterServiceVersion(h, operatorNamespace, operatorName)
+	checkConfigMapLockfile(h, operatorNamespace, operatorLockFile)
+	checkDeployment(h, operatorNamespace, operatorName, defaultDesiredReplicas)
+	checkClusterRoles(h, clusterRoles, false)
+
+	// should I create a new helper here? seems everyone else does but I am not sure
+	checkUpgrade(helper.New(),
+		operatorNamespace,
+		operatorName,
+		operatorName,
+		"route-monitor-operator-registry")
+
+	ginkgo.Context("Route Monitor Operator regression", func() {
+		ginkgo.Context("console", func() {
+			ginkgo.It("has all of the required resouces", func() {
+				var err error
+				// RouteMonitor is expected to be there as a selectorsyncset
+				// see https://github.com/openshift/managed-cluster-config/blob/master/deploy/osd-route-monitor-operator/100-openshift-console.console.RouteMonitor.yaml
+				_, err = h.Prometheusop().MonitoringV1().ServiceMonitors(consoleNamespace).Get(context.TODO(), consoleName, metav1.GetOptions{})
+				Expect(err).NotTo(HaveOccurred(), "Could not get console serviceMonitor")
+				_, err = h.Prometheusop().MonitoringV1().PrometheusRules(consoleNamespace).Get(context.TODO(), consoleName, metav1.GetOptions{})
+				Expect(err).NotTo(HaveOccurred(), "Could not get console prometheusRule")
+			})
+		})
+	})
+	ginkgo.Context("Route Monitor Operator integration test", func() {
+		var (
+			routeMonitorNamespace = h.CurrentProject()
+			routeMonitorName      = "routemonitor-e2e-test"
+		)
+		ginkgo.It("has the required prerequisites for testing", func() {
+			var err error
+
+			//Launch a pod, service && route as the targets of the ServiceMonitor and PrometheusRules objects
+			pod := helper.SamplePod(routeMonitorName, routeMonitorNamespace, "quay.io/brancz/prometheus-example-app:v0.2.0")
+			err = helper.CreatePod(pod, routeMonitorNamespace, h)
+			Expect(err).NotTo(HaveOccurred(), "Couldn't create user workload monitoring testing pod")
+			phase := h.WaitForPodPhase(pod, kubev1.PodRunning, 6, time.Second*15)
+			Expect(phase).To(Equal(kubev1.PodRunning), fmt.Sprintf("pod %s in ns %s is not running, current pod state is %v", routeMonitorName, routeMonitorNamespace, phase))
+
+			svc := helper.SampleService(8080, routeMonitorName, routeMonitorNamespace, routeMonitorName)
+			err = helper.CreateService(svc, h)
+			Expect(err).NotTo(HaveOccurred(), "Couldn't create user workload monitoring testing service")
```

Comment:
guess you want to create a service for RMO here. Makes sense, but the message is confusing.

Reply:
Yeah, I'll fix it :(